src/lib/libstrings.js

```diff
@@ -224,7 +259,22 @@ addToLibrary({
    * @param {string} str - JavaScript string to operator on
    * @return {number} Length, in bytes, of the UTF8 encoded string.
    */
+  $lengthBytesUTF8__deps: [
+#if TEXTENCODER == 2
+    '$UTF8Encoder',
+#endif
+  ],
```

Comment:
This can all be on single line and wrapped in `#if TEXTENCODER == 2`:

```
#if TEXTENCODER == 2
    $lengthBytesUTF8__deps: ['$UTF8Encoder'],
#endif
```